src/manageCalendars.js

```diff
@@ -0,0 +1,166 @@
+const commonModalAttributes = {
+  class: "mini",
+  closeIcon: true,
+  classContent: "centered",
+};
+
+/** Formantic Modal Designs */
+const formatModal = $.modal({
+  title: "Choose Calendar Upload Method",
+  ...commonModalAttributes,
+  content: `
+    <div>
+    <button id="upload-ical" class="ui right labeled icon button green">
+<i class="right arrow icon"></i>
+Upload an Ical Link
+</button>
+    <br/><br/>
+    <button  id="upload-manual" class="ui right labeled icon button green">
+<i class="right arrow icon"></i>
+Add Calendar Manually
+</button>
+    </div>`,
+});
+
+const addIcalModal = $.modal({
+  title: "Add Ical Link",
+  ...commonModalAttributes,
+  content: `
+    <div class="ui labeled input">
+    <div class="ui blue label">Name</div>
+    <input id="ical-name-input" type="text" placeholder="Eg. Sam">
+    </div>
+
+    <h5>Paste your Ical Link Here:</h5>
+
+    <div class="ui labeled input">
+
+    <div id = "ical-input" class="ui blue label">ICal Link</div>
+    
+    <input type="text" placeholder="<Ical Link>">
+    </div>
+    <br></br>
+    <div class="ui approve button green" id="setup-new-calendar-ical" >Done</div>`,
+});
+
+const addManualModal = $.modal({
+  title: "Add Ical Link",
+  ...commonModalAttributes,
+  content: `
+    <div class="ui labeled input">
+    <div class="ui blue label">Name</div>
+    <input id="manual-name-input" type="text" placeholder="Eg. Sam">
+    </div>
+
+    <h5>Add your Calendar Manually</h5>
+
+    <div class="ui labeled input">
+    <h4>Insert Manual Addition Here</h4></div>
+    <br></br>
+    <div class="ui approve button green" id="setup-new-calendar-manual">Done</div>`,
+});
+
+/** HTML Element Declarations */
+const title = document.getElementById("calendar-title");
+const icalName = document.getElementById("ical-name-input");
+const manualName = document.getElementById("manual-name-input");
+const dynamicSection = document.getElementById("dynamicTabs");
+
+const NON_DYNAMIC_NAV_ELEMENTS = 3;
+
+document
+  .getElementById("add-calendar-modal-open")
+  .addEventListener("click", addCalendar);
+document
+  .getElementById("view-combination-button")
+  .addEventListener("click", viewCombinedCalendar);
+document.getElementById("upload-ical").addEventListener("click", uploadIcal);
+document
+  .getElementById("upload-manual")
+  .addEventListener("click", uploadManual);
+document
+  .getElementById("setup-new-calendar-manual")
+  .addEventListener("click", setupNewManual);
+document
+  .getElementById("setup-new-calendar-ical")
+  .addEventListener("click", setupNewIcal);
+
+/** List of Uploaded Calendars */
+let calList = [];
```

Comment:
As a suggestion, I think if we were to have a variable which uses this _calList__ array to find it's size, that would be quite useful for other functions (outside this PR). My reasoning for this is that a function like the one calculating the colour of a given time period in the calendar could use this variable to simplify it considerably, rather than trying to count the total number of people input into the calendar itself at that time.

Reply:
Great Idea, this seems to be in scope for this pull request, will add that now!

Reply:
Awesome, thank you!